vowpalwabbit/parser.h

```diff
@@ -34,11 +45,12 @@ struct parser
   example* examples;
   uint64_t used_index;
   bool emptylines_separate_examples; // true if you want to have holdout computed on a per-block basis rather than a per-line basis
-  MUTEX examples_lock;
-  CV example_available;
-  CV example_unused;
-  MUTEX output_lock;
-  CV output_done;
+
+  std::mutex examples_lock;
```

Comment:
It's nice to include a comment on what's protected by this lock.

Reply:
Good idea.  The answer seems to be 'p->done', 'ec.in_use', p->begin_parsed_examples, p->end_parse_examples, and p->used_index.